src/backend/marsha/core/tests/test_views_lti_select.py

```diff
@@ -0,0 +1,147 @@
+"""Test the LTI select view."""
+from html import unescape
+import json
+import re
+from unittest import mock
+import uuid
+
+from django.test import TestCase
+from django.utils import timezone
+
+from ..factories import (
+    ConsumerSiteLTIPassportFactory,
+    DocumentFactory,
+    PlaylistFactory,
+    VideoFactory,
+)
+from ..lti import LTI
+from ..models import LTIPassport, Playlist
+
+
+# We don't enforce arguments documentation in tests
+# pylint: disable=unused-argument,too-many-lines
+
+
+class SelectLTIViewTestCase(TestCase):
+    """Test the select LTI view in the ``core`` app of the Marsha project."""
+
+    maxDiff = None
+
+    def test_views_lti_select_student(self):
+        """Error 403 raised if a student initiates the request."""
+        passport = ConsumerSiteLTIPassportFactory()
+        PlaylistFactory(consumer_site=passport.consumer_site)
+
+        # https://www.imsglobal.org/specs/lticiv1p0/specification
+        data = {
+            "context_id": passport.consumer_site.playlists.first().lti_id,
+            "roles": "student",
+        }
+
+        response = self.client.post(
+            "/lti/select/", data, HTTP_REFERER=passport.consumer_site
+        )
+        self.assertEqual(response.status_code, 403)
+
+    @mock.patch.object(LTI, "verify")
+    @mock.patch.object(LTI, "get_consumer_site")
+    def test_views_lti_select(self, mock_get_consumer_site, mock_verify):
+        """Validate the context passed to the frontend app for a LTI Content selection."""
+        passport = ConsumerSiteLTIPassportFactory()
+        resolutions = [144]
+        playlist = PlaylistFactory(consumer_site=passport.consumer_site)
+        VideoFactory(
+            id=1,
```

Comment:
Every id in marsha are UUID V4, you should use a real UUID

Reply:
Actually, don't force ids in tests.